quantuminspire/sdk/models/base_algorithm.py

```diff
@@ -0,0 +1,46 @@
+"""Module containing the Abstract Base Class for Algorithm classes."""
+from abc import ABC
+
+
+class BaseAlgorithm(ABC):
+    """A container object, used to store content and metadata for the algorithm."""
+
+    def __init__(self, platform_name: str, program_name: str) -> None:
+        self._platform_name = platform_name
+        self._program_name = program_name
+
+    @property
+    def program_name(self) -> str:
+        """Return the name of the quantum circuit.
```

Comment:
The docstring should be made generic, since this is not only for quantum cirquits.

Reply:
Done